### DISCUSSION TITLE
Confusion on jj sparse

### DISCUSSION THREAD

**Discussion by iljabauer** [jj-vcs/jj#4658] [Q&A] [answered]
Hi,

I'm pretty new to jj but used it manage some of my repos so far without problems. Thank you for jj 🙏

Now I have a giant repository and I try to get it to work. I cannot clone it with git normaly, so I always do:
```
git clone --filter=blob:none --no-checkout <repo>
git sparse-checkout init --cone
```

No I thought I can replace git sparce commands with jj sparse, so I do:
```
git clone --filter=blob:none --no-checkout <repo>
jj git init --colocate
jj sparse set --clear --add <path>
```

But nothing happens, folder is empty and `jj status` shows all files as deleted, do I miss something?

**Comment by martinvonz** [comment, marked as answer]
I wasn't familiar with the `--no-checkout` argument. As the name suggests, it skips the checkout. I would have expected that it left HEAD on an unborn branch or something, but it seems that it simply leaves the index empty, i.e. making its diff compared to HEAD be to remove all files. That's not very friendly IMO, but I understand that Git is limited by not having a virtual root commit.

More importantly, jj doesn't support partial clone (the `--filter=blob:none` stuff) yet, so it won't work even if you skip `--no-checkout` (or restore the deleted files).

Also, there's no syncing of patterns between jj's and git's sparse checkouts, so you would have to keep them in sync manually if you use a colocated workspace.

So, you will have to do `jj git clone [--shallow <depth>] <repo>` and then `jj sparse set --clear --add <path>` to get it to work, but that will of course not give you a partial clone, so maybe it's still too large or too slow for you.

**Reply by iljabauer** [reply]
Got it! Thank you for your help! I'll try 👍
